tests/Game.class.test.js

```diff
@@ -0,0 +1,916 @@
+
+        game2048.start();
+        game2048.moveLeft();
+
+        const state = game2048.getState();
+
+        state.forEach((row, rowIndex) => {
+          const cursor = initialState[rowIndex].find((cell) => cell !== 0);
+
+          expect(checkIsCorrectlyAligned(row, 'start', cursor)).toBe(true);
+        });
+      });
+
+      it('should add a random cell with 2 or 4 at an empty position after the move', () => {
+        const initialState = [
+          [8, 0, 16, 0],
+          [0, 16, 0, 32],
+          [8, 0, 32, 0],
+          [0, 16, 0, 64],
+        ];
+
+        const game2048 = new Game(cloneState(initialState));
+
+        game2048.start();
+
+        const stateAfterStart = [...game2048.getState()];
+
+        game2048.moveLeft();
+
+        const stateAfterMove = game2048.getState();
+
+        const addedCells = stateAfterMove
+          .reduce((acc, row, rowIndex) => {
+            const countBeforeMove = stateAfterStart[rowIndex].filter((cell) => cell !== 0).length;
+
+            if (countBeforeMove === 4) {
+              return acc;
+            }
+
+            const newCells = row.slice(countBeforeMove).filter((cell) => cell !== 0);
+
+            return [
+              ...acc,
+              ...newCells,
+            ];
+          }, []);
+
+        expect(addedCells.length).toBe(1);
+        expect([2, 4].includes(addedCells[0])).toBe(true);
+      });
+
+      it('should stack cells to the left without merging', () => {
+        const initialState = [
+          [0, 16, 0, 8],
+          [8, 0, 16, 0],
+          [0, 8, 0, 32],
+          [32, 0, 8, 0],
+        ];
+
+        const game2048 = new Game(cloneState(initialState));
+
+        game2048.start();
+        game2048.moveLeft();
+
+        const state = game2048.getState();
+
+        state.forEach((row, rowIndex) => {
+          const initialRow = initialState[rowIndex];
+          const nonEmptyCells = initialRow.filter((cell) => cell !== 0);
+          const cursor = nonEmptyCells.slice(-1)[0];
+
+          expect(checkIsCorrectlyAligned(row, 'start', cursor)).toBe(true);
+
+          nonEmptyCells.forEach((cell) => {
+            expect(row.includes(cell)).toBe(true);
+          });
+        });
+      });
+
+      it('should merge cells with the same value', () => {
+        const initialState = [
+          [16, 8, 0, 0],
+          [0, 0, 8, 8],
+          [0, 8, 8, 0],
+          [8, 0, 0, 16],
+        ];
+
+        const game2048 = new Game(cloneState(initialState));
+
+        game2048.start();
+        game2048.moveLeft();
+
+        const state = game2048.getState();
+
+        state.forEach((row) => {
+          expect(row.includes(16)).toBe(true);
+        });
+      });
+
+      it('should merge multiple cells with the same value to the left', () => {
+        const initialState = [
+          [8, 8, 16, 16],
+          [16, 16, 8, 8],
+          [32, 32, 8, 8],
+          [16, 16, 16, 16],
```

Comment:
this line should go in a separate test 
that 3 element in a row are not merged all together

Reply:
Reasonable, will add it